source/settings/plugin/sql-cassandra.rst

```diff
@@ -0,0 +1,374 @@
+
+   Read consistency if primary consistency fails.
+
+
+.. dovecot_core:setting:: request_timeout
+   :default: 60s
+   :domain: sql-cassandra
+   :values: @time_msecs
+
+   How long to wait for a query to finish.
+
+
+.. dovecot_core:setting:: ssl_ca
+   :domain: sql-cassandra
+   :values: @string
+
+   Path to SSL certificate authority file to use to validate peer certificate.
+
+
+.. dovecot_core:setting:: ssl_cert_file
+   :domain: sql-cassandra
+   :values: @string
+
+   Path to a certificate file to use for authenticating against the remote
+   server.
+
+
+.. dovecot_core:setting:: ssl_private_key_file
+   :domain: sql-cassandra
+   :values: @string
+
+   Path to private key matching
+   :dovecot_core:ref:`sql-cassandra;ssl_cert_file` to use for authenticating
+   against the remote server.
+
+
+.. dovecot_core:setting:: ssl_verify
+   :default: none
+   :domain: sql-cassandra
+   :values: none, cert, cert-ip, cert-dns
+
+   Configure the peer certificate validation method.
+
+   Options:
+
+   ``none``
+     Disables validation.
+
+   ``cert``
+     Validate that the certificate is valid.
+
+   ``cert-ip``
+     Validate that the certificate is valid and has Common Name or Subject
+     Alternate Name for the IP address.
+
+   ``cert-dns``
+      Validate that the certificate is valid and has Common Name or Subject
+      Alternate Name that matches PTR resource record for the server's IP
+      address.
+
+
+.. dovecot_core:setting:: user
+   :domain: sql-cassandra
+   :values: @string
+
+   Username for authentication.
+
+
+.. dovecot_core:setting:: version
+   :default: !Depends on driver version.
+   :domain: sql-cassandra
+   :values: v3, v4, v5
+
+   Cassandra driver version to use. It is good idea to specify this to avoid
+   warnings about version handshake.
+
+   .. note:: If you want to use server-side prepared statements, you need to
+             use at least ``v3``.
+
+
+.. dovecot_core:setting:: warn_timeout
+   :default: 5s
+   :domain: sql-cassandra
+   :values: @time_msecs
+
+   Emit warning if query takes longer than this.
+
+
+.. dovecot_core:setting:: write_consistency
+   :default: local-quorum
+   :domain: sql-cassandra
+   :seealso: @dictmap_cassandra_quorum_configuration
+   :values: !:ref:`cassandra_consistency`
+
+   Write consistency when updating or inserting to the database.
+
+
+.. dovecot_core:setting:: write_fallback_consistency
+   :default: local-quorum
+   :domain: sql-cassandra
+   :seealso: @dictmap_cassandra_fallback_consistency
+   :values: !:ref:`cassandra_consistency`
+
+   Write consistency when updating or inserting to the database fails with
+   primary consistency.
+
+
+.. _cassandra_consistency:
+
+Cassandra Consistency Values
+============================
+
+Consistency levels in Cassandra can be configured to manage availability versus data accuracy.
+
+For read consistency following values are supported
+---------------------------------------------------
```

Comment:
Isn't it a bit weird to have a full sentence as a title? Shouldn't it instead be more like:

```
Read Consistency
----------------

 For read consistency the following values are supported:
```

And same for write/delete consistency.